ignite/internal/plugin/execute.go

```diff
@@ -35,11 +35,12 @@ func Execute(ctx context.Context, path string, args []string, options ...plugin.
 	if err != nil {
 		// Extract the rpc status message and create a simple error from it.
 		// We don't want Execute to return rpc errors.
-		err = errors.New(status.Convert(err).Message())
+		return "", errors.New(status.Convert(err).Message())
 	}
 	// NOTE(tb): This pause gives enough time for go-plugin to sync the
 	// output from stdout/stderr of the plugin. Without that pause, this
 	// output can be discarded and absent from buf.
 	time.Sleep(100 * time.Millisecond)
+	plugins[0].KillClient()
```

Comment:
We defer killing the client above, why is this one needed?

Reply:
Okay, then we can remove deferring killing the client right? As it will be already killed here.

Reply:
We have 2 error handlers before the last return. We need the defer

Reply:
It seems that adding the `KillClient` call might make the `time.Sleep` call redundant 🤔, so it might be possible to remove the `time.Sleep`. Killing the client should try to stop it gracefully, so buffers should be flushed properly if the plugin client is running without issues.

Maybe you could add a TODO? If it can't be properly tested to be sure in fact the `time.Sleep` is redundant now.

Reply:
Add into a issue for we can check it later

https://github.com/ignite/cli/issues/4101